File: front-end/app/src/app/commons/product.ts
import { ProductCategory } from "./product-category";

export class Product {
    constructor(
        public id: number,
        public name: string,
        public sku: string,
        public description: string,
        public price: number,
        public imageUrl: string,
        public isActive: boolean,
        public unitsInStock: number,
        public createdAt: Date,
        public updatedAt: Date,
        public category:ProductCategory
    ){}
}

File: back-end/app/src/main/java/com/ecommerce/app/model/Product.java
package com.ecommerce.app.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.math.BigDecimal;
import java.time.LocalDateTime;

@Entity
@Table(name = "product")
@Data
public class Product {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String name;

    private String sku;

    private String description;

    @Column(name = "unit_price")
    private BigDecimal price;

    private String imageUrl;

    @Column(name = "active")
    private Boolean isActive;

    private Integer unitsInStock;

    @CreationTimestamp
    @Column(name = "date_created")
    private LocalDateTime createdAt;

    @UpdateTimestamp
    @Column(name = "last_updated")
    private LocalDateTime updatedAt;

    @ManyToOne
    @JoinColumn(name = "category_id", nullable = false)
    private ProductCategory category;
}
